src/HDF5.jl

```diff
@@ -1340,29 +1340,35 @@ function create_external_dataset(parent::Union{File,Group}, name::AbstractString
 end
 
 """
-    do_write_chunk(dataset::Dataset, offset, chunk_bytes::Vector{UInt8}, filter_mask=0)
+    do_write_chunk(dataset::Dataset, offset, chunk_bytes::AbstractArray, filter_mask=0)
 
 Write a raw chunk at a given offset.
+`chunk_bytes` is an AbstractArray that can be converted to a pointer, Ptr{Cvoid}.
 `offset` is a 1-based list of rank `ndims(dataset)` and must fall on a chunk boundary.
 """
-function do_write_chunk(dataset::Dataset, offset, chunk_bytes::Vector{UInt8}, filter_mask=0)
+function do_write_chunk(dataset::Dataset, offset, chunk_bytes::AbstractArray, filter_mask=UInt32(0))
```

Comment:
Why add the `UInt32` here? This dispatches internally to https://github.com/JuliaIO/HDF5.jl/blob/master/src/api/functions.jl#L594 which takes care of the conversion to UInt32.

Reply:
The `filter_mask` is a 32 bit mask. You can enable or disable the 32-bits to indicate which filters you have skipped when writing the chunk directly.
https://portal.hdfgroup.org/display/HDF5/H5D_WRITE_CHUNK

All we are changing here is the *default* value, which is rarely changed explicitly. It does not make sense to create a `Int64` `0` and go through the trouble of truncating it every time this is used by default.

Importantly, one needs to use the `UInt32` path when reading raw chunks and rewriting them. The corresponding `do_read_chunk` will always return a `UInt32`.
https://github.com/JuliaIO/HDF5.jl/blob/eeb227db717ccdf0e9ad2e6c2853323ff8810f63/src/HDF5.jl#L1372-L1378

Thus if you used `getindex` followed by `setindex!` you would end up passing the `filter_mask` as a `UInt32`, thus I expect the code path with `UInt32` to be the most frequently used one.